xla/stream_executor/stream_executor_internal.h

```diff
@@ -140,6 +140,10 @@ class CommandBufferInterface {
                                            const DeviceMemoryBase& src,
                                            uint64_t size) = 0;
 
+  // Adds a device memory allocation node to the command buffer.
+  virtual tsl::Status MemallocOnDevice(uint64_t size,
```

Comment:
I think we have to use the original way to return address. Because `Allocate` node's update operation does not do the re-allocation, so it does not return value, but we should also assume it `OK` status. If we use `tsl::StatusOr<DeviceMemoryBase` as `record` return, we have no way to distinguish when its value is actually valid.

Reply:
One problem is that `CommandBufferCmdSequence` created only once and reused multiple times on different StreamExecutors, so in multi-gpu case you'll assign device memory only first time, and then will be retunng nullptr.

For mem alloc node in update stage you need to `cuGraphMemAllocNodeGetParams` and return a memory ptr owned by the node itself